MovieQuiz/Services/StatisticService.swift

```diff
@@ -56,10 +59,13 @@ final class StatisticServiceImplementation: StatisticService {
     
     func store(correct count: Int, total amount: Int) {
         gamesCount += 1
+        
         let newGame: GameRecord = GameRecord(correct: count, total: amount, date: Date().dateTimeString)
         if bestGame < newGame {
             bestGame = newGame
         }
-        totalAccuracy += Double(count)/Double(amount)
+        
+        //чтобы высчитать мы берём прошлое значение, умножаем на количество игр, которое было для этого-прошлого значения, так мы получаем общую сумму соотношений прошлых игр; затем добавляем соотношение новой игры и делим уже на текущие значение соотношений(количество сыгранных игр)
+        totalAccuracy = (totalAccuracy*Double(gamesCount-1)+Double(count)/Double(amount))/Double(gamesCount)
```

Comment:
блин :) а чо так сложно-то)

Давай проверим на таком тесте:
6/10 6/10 9/10 - 70%

(0 * 0 + 6/10 ) / 1 - 60%
(0,6 * 1 + 0.6) / 2 - 60%
(0,6 * 2 + 0.9) / 3 - 70%

Звучит неплохо! 
В будущем не забудь такой автотест написать. 

Что по коду:

Обычно принято вокруг операторов пробелы расставлять:
```suggestion
        totalAccuracy = (totalAccuracy * Double(gamesCount -  1) + Double(count) / Double(amount)) / Double(gamesCount)
```

А еще лучше – промежуточные вычисления делать. 
Твоя формула полагается на то, что все игры сыграны на одинаковом количестве вопросов (так по ТЗ), но для расширения такой подход не очень гибкий. Лучше сохранять количество верных ответов за всю историю и количество отвеченных вопросов. Так будет работать немного погибче.  Но твое решение выглядит рабочим, поэтому можно оставить :)

Reply:
Да, действительно неправильно для разного числа вопросов в раунде; поменял ещё раз, я просто зациклился на реализации через сеттер для totalAccuracy, но без него код намного проще и гибче)